### PR TITLE
Integration of CoreBundle

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
Todo:
* [x] Take old admin clases
* [x] Use DI factory for configuration
* [x] Admin Web-Tests

### CODE REVIEW

File: src/DependencyInjection/Factory/CoreAdminFactory.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+
+    }
```
Comment (wouterj) [inline src/DependencyInjection/Factory/CoreAdminFactory.php]: if you don't need process, please remove it (and its interface)

File: src/DependencyInjection/CmfSonataAdminIntegrationExtension.php [hunk 2]
```diff
@@ -40,14 +40,15 @@ public function load(array $configs, ContainerBuilder $container)
 
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
 
-        $this->loadBundles($config['bundles'], $loader, $container);
+        $this->loadByBundles($config['bundles'], $loader, $container);
     }
 
-    private function loadBundles(array $config, XmlFileLoader $loader, $container)
+    private function loadByBundles(array $config, XmlFileLoader $loader, ContainerBuilder $container)
```
Comment (wouterj) [inline src/DependencyInjection/CmfSonataAdminIntegrationExtension.php]: I don't like this rename

File: src/DependencyInjection/CmfSonataAdminIntegrationExtension.php [hunk 3]
```diff
@@ -40,14 +40,15 @@ public function load(array $configs, ContainerBuilder $container)
...
     {
         foreach ($this->factories as $name => $factory) {
             if ($this->isConfigEnabled($container, $config[$name])) {
                 $factory->create($config[$name], $container, $loader);
+
```
Comment (wouterj) [inline src/DependencyInjection/CmfSonataAdminIntegrationExtension.php]: should be reverted

File: src/Resources/config/core.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+        <service
+                id="cmf_sonata_admin_integration.core.admin_extension.child"
+                class="Symfony\Cmf\Bundle\SonataAdminIntegrationBundle\Admin\Core\Extension\ChildExtension"
+                public="false">
+            <tag name="sonata.admin.extension"/>
+        </service>
+
+        <service
+                id="cmf_sonata_admin_integration.core.admin_extension.publish_workflow.publishable"
```
Comment (wouterj) [inline src/Resources/config/core.xml]: s/admin_extension/extension (same above)

File: src/DependencyInjection/CmfSonataAdminIntegrationExtension.php (hunk 2 above)
Comment (ElectricMaxxx) [inline src/DependencyInjection/CmfSonataAdminIntegrationExtension.php]: sorry, was a result of a merge conflict, had an almost equal function before i rebased on master with your factories.

File: src/DependencyInjection/Factory/CoreAdminFactory.php [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $config, ContainerBuilder $container, XmlFileLoader $loader)
+    {
+        $loader->load('core.xml');
+
+        $container->setParameter('cmf_sonata_admin_integration.core.form_group', $config['form_group']);
+
+        if (!isset($config['extensions'])) {
```
Comment (wouterj) [inline src/DependencyInjection/Factory/CoreAdminFactory.php]: this is useless, it's always set.

**Comment by ElectricMaxxx** [comment]
@wouterj you didn't have those breaking build for php 7.0 and symfony 3.0 (deprecations) on routing and core, right?

**Comment by ElectricMaxxx** [comment]
i think the failing build is a phpunit or phpunit-bridge problem. The failing one calls a new composer update with a version change -> dev-master came into the field.

**Comment by ElectricMaxxx** [comment]
It can not be the phpunit version as i have the same that fails on green one too. The phpunit version seems to correlate to the php version. But for some reasons the simple-phpunit scripts seems to change the dependencies befor testing on the failing build. I think the reason is: i get dev-master of phpunit-bridge instead of 3.2.

**Comment by ElectricMaxxx** [comment]
Got it. Thanks to @wouterj He describes, that the major point is the amount of deprecation warning. So using weak for the broken test solves my problem.

**Comment by ElectricMaxxx** [comment]
@symfony-cmf/contributors ready to review, please.

File: .travis.yml [hunk 6]
```diff
@@ -26,7 +26,7 @@ matrix:
     - php: 5.5
       env: COMPOSER_FLAGS="--prefer-lowest" SYMFONY_VERSION=2.8.* SYMFONY_DEPRECATIONS_HELPER=weak
     - php: 7.0
-      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.*
+      env: COMPOSER_FLAGS="--prefer-stable" SYMFONY_VERSION=3.0.* SYMFONY_DEPRECATIONS_HELPER=weak
```
Comment (wouterj) [inline .travis.yml]: this is not a lowest dep test. You should remove this one and change the numer on line 19 to be the number of deprecations that is happening at the moment.

This means we get a failing build when we introduce new deprecated usages. As much usages are caused by Sonata, we can ignore them.

**Comment by ElectricMaxxx** [comment]
@wouterj ok for you too? (still got a declined review from you side)

**Comment by ElectricMaxxx** [comment]
the removement is merged, so lets do it here too.

**Pull request merged**